### PR TITLE
Astro featured articles

### PR DESCRIPTION
**Pull request by omar-h-omar**
I've added a portrait flag to the article drafts to tell us if the thumbnail is landscape or portrait. I've tried to make the design as modular as possible. If you spot anything that I can improve just let me know. I haven't pushed the s3URL file like you requested. I'm assuming it will be in your ticket. Finally I can't see a way to test this without being able to control what articles are coming back from prisma. If you have any suggestions let me know but I've commented out the testing job for now.

### CODE REVIEW

**Comment by omar-h-omar** [comment]
Unrealed to this: @joe2k01 should we turn off that spelling check? It's messy and not helpful

File: readers-frontend/src/components/utils/articles.ts [hunk 1]
```diff
@@ -0,0 +1,57 @@
+import { Prisma } from "@prisma/client";
+import prisma from "../../prisma";
+import { s3URL } from "./s3Url";
```
Comment (joe2k01) [inline readers-frontend/src/components/utils/articles.ts:3]: It should be s3URL, case sensitive file systems will break here (mine did)

**Review by joe2k01 — CHANGES_REQUESTED** [review]

File: readers-frontend/dist/index.html [hunk 2]
```diff
@@ -83,64 +81,180 @@
...
+		
+
+<!-- 1 article --><div class="flex w-full gap-2.5">
+    <div class="max-w-[50%] border-r-2 pr-2.5 border-gray-300">
+      
+
+
+<div class="h-full text-ellipsis overflow-hidden astro-Q7KXOHAS">
+  <div class="flex astro-Q7KXOHAS" style="--categoryColor: #000000;">
+      <a class="w-1/2 h-full  astro-Q7KXOHAS" href="/2022/09/04/featuredArticles[0].articles_slug">
```
Comment (joe2k01) [inline readers-frontend/dist/index.html:94]: I don't think this is meant to be a string?
Shouldn't it be
```js
`/2022/09/04/${featuredArticles[0].articles_slug}`
```

There's a couple more occurrences of the same

**Review by joe2k01 — CHANGES_REQUESTED** [review]

File: readers-frontend/dist/index.html (hunk 2 above)
Comment (joe2k01) [reply readers-frontend/dist/index.html:94]: I mean I picked one from the dist folder, but that's coming from the source

File: readers-frontend/src/components/FeaturedArticles.astro [hunk 3]
```diff
@@ -0,0 +1,380 @@
...
+        hex = x.charCodeAt(i).toString(16);
+        result += ("000"+hex).slice(-4);
+    }
+
+    return result
+}
+const mainCategories: (categories | null)[] = await Promise.all(featuredArticles.map(article => getMainArticleCategory(article.articles_categories.split(',').map(Number))))
+---
+{!hasPortraitArticles &&
+  <!-- 4 or more Articles -->
```
Comment (joe2k01) [inline readers-frontend/src/components/FeaturedArticles.astro:43]: Can we have these as Astro comments so they don't all show up in the generated html?

**Review by joe2k01 — CHANGES_REQUESTED** [review]

File: readers-frontend/src/components/FeaturedArticles.astro (hunk 3 above)
Comment (omar-h-omar) [reply readers-frontend/src/components/FeaturedArticles.astro]: Ahh I didn't know about that. I fixed it now :)

File: readers-frontend/dist/index.html (hunk 2 above)
Comment (omar-h-omar) [reply readers-frontend/dist/index.html]: Yup You're right! I fixed it now

File: readers-frontend/src/components/utils/articles.ts (hunk 1 above)
Comment (omar-h-omar) [reply readers-frontend/src/components/utils/articles.ts]: Updated!

**Review by joe2k01 — APPROVED** [review]

**Pull request merged**